### PR TITLE
[WIP][MIG]school_transport: Migration to v14.

### PR DESCRIPTION
**Pull request by ForamSerpentCS**

### CODE REVIEW

File: school_transport/models/transport.py [hunk 1]
```diff
@@ -236,22 +262,23 @@ class StudentStudent(models.Model):
...
-            if trans_student:
-                trans_student.write({'active': False})
+            trans_student_rec = self.env['transport.participant'].search([
+                                                    ('name', '=', rec.id)])
+            trans_regi_rec = self.env['transport.registration'].search([
+                                                ('part_name', '=', rec.id)])
+            if trans_regi_rec:
+                trans_regi_rec.write({'state': 'cancel'})
+            if trans_student_rec:
+                trans_student_rec.write({'active': False})
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: trans_student_rec.active = False

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

File: school_transport/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# See LICENSE file for full copyright and licensing details.
+
+{
+    "name": "Transport Management",
```
Comment (JayVora-SerpentCS) [inline school_transport/__manifest__.py]: Transport Management - A Part of Education ERP

File: school_transport/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "version": "14.0.1.0.0",
+    "author": "Serpent Consulting Services Pvt. Ltd.",
+    "website": "http://www.serpentcs.com",
+    "license": "AGPL-3",
+    "category": "School Management",
+    "complexity": "easy",
+    "summary": "A Module For Transport & Vehicle Management In School",
+    "depends": ["school"],
```
Comment (JayVora-SerpentCS) [inline school_transport/__manifest__.py]: check for the FLEET compatability. It isa nearfit now.
@AnuKPatel @ForamSerpentCS

File: school_transport/models/transport.py [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+        """Method to change state to close."""
+        self.state = "close"
+
+    @api.model
+    def participant_expire(self):
+        """Schedular to change in participant state when registration date.
+
+        is over.
+        """
+        current_date = fields.datetime.today()
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: fields.Date.today()

File: school_transport/models/transport.py [hunk 5]
```diff
@@ -0,0 +1,623 @@
...
+        help="Student transportation",
+    )
+
+    def set_alumni(self):
+        """Override method to make record of student transport active false.
+
+        when student is set to alumni state.
+        """
+        for rec in self:
+            trans_student_rec = self.env["transport.participant"].search(
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: Define self.env objects outside the for loop

File: school_transport/models/transport.py [hunk 6]
```diff
@@ -0,0 +1,623 @@
...
+                [("name", "=", rec.id)]
+            )
+            trans_regi_rec = self.env["transport.registration"].search(
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: same as above

File: school_transport/models/transport.py [hunk 7]
```diff
@@ -0,0 +1,623 @@
...
+                [("part_name", "=", rec.id)]
+            )
+            if trans_regi_rec:
+                trans_regi_rec.write({"state": "cancel"})
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: trans_regi_rec.state = "cancel"

File: school_transport/models/transport.py [hunk 8]
```diff
@@ -0,0 +1,623 @@
...
+    name = fields.Many2one(
+        "student.transport",
+        "Transport Root Name",
+        domain=[("state", "=", "open")],
+        required=True,
+        help="Enter transport root name",
+    )
+    part_name = fields.Many2one(
+        "student.student", "Student Name", required=True, help="Student Name"
+    )
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py:347]: Did not understand the reason of the field name part_name. Here field name should be student_id,

File: school_transport/models/transport.py [hunk 9]
```diff
@@ -0,0 +1,623 @@
...
+        readonly=True,
+        help="Start Date of registration",
+        default=fields.Date.context_today,
+    )
+    reg_end_date = fields.Date(
+        "Registration End Date",
+        readonly=True,
+        help="Start Date of registration",
+    )
+    for_month = fields.Integer("Registration For Months")
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: Change the field name as registration_month

File: school_transport/models/transport.py [hunk 10]
```diff
@@ -0,0 +1,623 @@
...
+        help="""State of the transport
+                             registration form""",
+    )
+    vehicle_id = fields.Many2one(
+        "fleet.vehicle",
+        "Vehicle No",
+        required=True,
+        help="Enter transport vehicle",
+    )
+    m_amount = fields.Float(
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: the field name should be monthly_amount

File: school_transport/models/transport.py [hunk 11]
```diff
@@ -0,0 +1,623 @@
...
+            journal = invoice.journal_id
+            acct_journal_id = journal.default_account_id.id
+            account_view_id = self.env.ref("account.view_move_form")
+            line_vals = {
+                "name": "Transport Fees",
+                "account_id": acct_journal_id,
+                "quantity": rec.for_month,
+                "price_unit": rec.m_amount,
+            }
+            invoice.write({"invoice_line_ids": [(0, 0, line_vals)]})
```
Comment (Trivedi-Vacha-SerpentCS) [inline school_transport/models/transport.py]: invoice.invoice_line_ids = [(0, 0, line_vals)]

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

**Comment by JayVora-SerpentCS** [comment]
@ForamSerpentCS see the latest v14 codes. There are smart one line codes whch will save time.

**Review by Trivedi-Vacha-SerpentCS — APPROVED** [review]

**Pull request merged**